pp_hot.c

```diff
@@ -3107,9 +3106,13 @@ PP(pp_match)
                         "start=%ld, end=%ld, s=%p, strend=%p, len=%" UVuf,
                         (long) i, (long) RXp_OFFS(prog)[i].start,
                         (long)RXp_OFFS(prog)[i].end, s, strend, (UV) len);
-                sv_setpvn(*SP, s, len);
-                if (DO_UTF8(TARG) && is_utf8_string((U8*)s, len))
-                    SvUTF8_on(*SP);
+                PUSHs(newSVpvn_flags(s, len,
+                    (DO_UTF8(TARG) && is_utf8_string((U8*)s, len))
+                    ? SVf_UTF8|SVs_TEMP
+                    : SVs_TEMP)
+                );
+            } else {
+                PUSHs(sv_newmortal());
```

Comment:
The is_utf8_string() is no longer needed here.

It was added in cce850e4f65 to support the \C regexp class, but that was removed in perl 5.24.

Either I could remove it and you rebase on top, or you could (preferably as a separate commit with an appropriate subject.)

Reply:
I'm happy to add a commit that removes it. I'll try to do that later today.

Reply:
Just to clarify, will a separate commit at the tail of this PR do, or are you looking for a separate PR followed by a rebase here when that separate PR is merged?

Reply:
I've opened #19154, will rebase this PR once it gets merged.